File: frontend/src/services/prescriptionService.js
import api from './api'

const prescriptionService = {
  // Admin: all prescriptions (paginated + searchable)
  getAll:           (params)     => api.get('/prescriptions',                    { params }),

  // By ID
  getById:          (id)         => api.get(`/prescriptions/${id}`),

  // Doctor: create
  create:           (data)       => api.post('/prescriptions',                   data),

  // Doctor: update
  update:           (id, data)   => api.put(`/prescriptions/${id}`,              data),

  // Doctor: own prescriptions
  getMy:            (params)     => api.get('/prescriptions/my',                 { params }),

  // By patient profile ID
  getByPatient:     (id, params) => api.get(`/prescriptions/patient/${id}`,      { params }),

  // By doctor profile ID
  getByDoctor:      (id, params) => api.get(`/prescriptions/doctor/${id}`,       { params }),

  // By appointment ID
  getByAppointment: (id, params) => api.get(`/prescriptions/appointment/${id}`,  { params }),

  // Admin: delete
  delete:           (id)         => api.delete(`/prescriptions/${id}`),
}

export default prescriptionService

File: backend/src/main/java/com/medicoai/service/PrescriptionService.java
package com.medicoai.service;

import com.medicoai.dto.request.PrescriptionRequest;
import com.medicoai.dto.response.PrescriptionResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface PrescriptionService {

    /** Doctor creates a prescription for a patient. doctorUserId = logged-in doctor's user ID. */
    PrescriptionResponse create(Long doctorUserId, PrescriptionRequest request);

    /** Get prescription by ID. */
    PrescriptionResponse getById(Long id);

    /** All prescriptions for a patient (paginated). */
    Page<PrescriptionResponse> getByPatient(Long patientId, Pageable pageable);

    /** All prescriptions written by a doctor (paginated). */
    Page<PrescriptionResponse> getByDoctor(Long doctorId, Pageable pageable);

    /** All prescriptions written by a doctor resolved from their user ID (for /my endpoint). */
    Page<PrescriptionResponse> getByDoctorUserId(Long doctorUserId, Pageable pageable);

    /** All prescriptions linked to an appointment. */
    Page<PrescriptionResponse> getByAppointment(Long appointmentId, Pageable pageable);

    /** Admin: paginated list with optional search. */
    Page<PrescriptionResponse> getAll(String search, Pageable pageable);

    /** Update an existing prescription (doctor only). */
    PrescriptionResponse update(Long id, PrescriptionRequest request);

    /** Delete a prescription (admin only). */
    void delete(Long id);
}
